dap_aria_mapping/analysis/app/pages/1_Horizon_Scanner.py

```diff
@@ -0,0 +1,307 @@
+import streamlit as st
+from PIL import Image
+import altair as alt
+from nesta_ds_utils.viz.altair import formatting
+from dap_aria_mapping import PROJECT_DIR
+from dap_aria_mapping.getters.app_tables.horizon_scanner import volume_per_year
+from dap_aria_mapping.getters.taxonomies import get_topic_names
+import polars as pl
+import pandas as pd
+from typing import List, Tuple
+
+formatting.setup_theme()
+
+PAGE_TITLE = "Horizon Scanner"
+
+IMAGE_DIR = f"{PROJECT_DIR}/dap_aria_mapping/analysis/app/images"
+
+#icon to be used as the favicon on the browser tab
+icon = Image.open(f"{IMAGE_DIR}/hs_icon.ico")
+
+# sets page configuration with favicon and title
+st.set_page_config(
+    page_title=PAGE_TITLE, 
+    layout="wide", 
+    page_icon=icon
+)
+
+@st.cache_data
+def load_overview_data() -> Tuple[pl.DataFrame, pl.DataFrame, List[str]]:
+    """loads in the volume per year chart and does initial formatting that is not impacted by filters
+    caches results so the data is not loaded each time a filter is run
+
+    Returns:
+        pl.DataFrame: total patents/publications per domain/area/topic per year, with names
+        pl.DataFrame: same as above, but patent/publication counts are melted to long form
+        List: unique domain names in dataset
+    """
+    volume_data = volume_per_year()
+
+    #add total document count as count of patents and publications combined
+    volume_data = volume_data.with_columns(
+        (pl.col('patent_count') + pl.col('publication_count')).alias('total_docs'),
+        pl.col('year').round(0)
+    )
```

Comment:
What is the best practice on pre-calculating everything and just loading ready-made tables, versus doing some calculations in here? (I appreciate this is very simple addition, so super quick).

Reply:
Yep good suggestion - I just did that for short-term efficiency as the pipeline takes longer to run and I realised after the fact that I needed it. But just moved this and the topic naming into the pipeline rather than on the load of the app.